### ISSUE TITLE
Update Copyright notices

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2883]
From: **How to Use GNU Licenses for Your Own Software** https://www.gnu.org/licenses/gpl-howto.html we are required to include a "Copyright" notice within code files:

- "Give each file the proper copyright notices. ...": proposed update target for this issue.
- "Put a license notice in each file.": we may have some inadvertent omissions on this front - to be address in this issue also.

This maintenance is overdue, we have in part attended piecemeal to parts of this requirement. But mainly regarding updating dates. It is proposed that we now establish the new norm (prior to our next Stable release) re these notices.

# Proposals

1. Continuity of Copyright attribution. Rockstor's project founder, and initial maintainer, @schakrava established the project norm re Copyright attribution. They, along with @phillxnet (current maintainer by permission from @schakrava) & @FroggyFlox (current co-maintainer via permission from @phillxnet) form the current Admin panel of [The Rockstor Project](https://opencollective.com/the-rockstor-project). As such there is continuity re natural person (proposed persons) - along with an increased robustness re removal of single entity re natural persons able to raise Copyright infringement claims. We also, in this proposal, remove the now outdated `Inc` reference which would be replaced by our current community/backer/fiscal-host reference of "The Rockstor Project". Ergo: natural person/s continuity regarding the Copyright representative/s for the project as it stands now: a non-profit Open Collective fiscally hosted by [Open Collective Europe](https://opencollective.com/europe) where the original natural person @schakrava persists by way of the assumed natural persons constituting The Rockstor Projects's administration: @phillxnet @FroggyFlox @schakrava.

2. We normalise all dates to be the current year only (we constantly release and so this year is our 'latest' release). This keeps things simple for future updates. And each update constitutes all files: modified or otherwise. From the above GNU reference doc:

> The copyright notice should include the year in which you finished preparing the release. ...

3. Explicitly indicate the Copyright as a "Joint Work":

From: https://www.copyright.gov/title17/title17.pdf, a US reference but there is understood to be a great deal of cross compatibility re world Copyright law (hence its use in Copyleft Licensing such as our GPL-3.0-or-later in this repo):

> A “joint work” is a work prepared by two or more authors with the intention
> that their contributions be merged into inseparable or interdependent parts of
> a unitary whole.

Given the proposed Copyright transition from effectively a single natural person (via an Inc) to our current Open Collective admin panel (all current and past maintainers), consisting of several persons: we indicate a joint representation there-in. The original copyright assignment aim in this project, and one common across many open source projects, is to avoid the like of squabbles, project-paralysis,  hijacking available resources with make-work (read re-writes) re who-wrote what part of what line at what time. We instead have explicitly assigned Copyright: in the proposed case to the current admin panel.

I.e.: Replace the likes of (we have a few variations of this):
```
Copyright (c) 2012-2020 RockStor, Inc. <http://rockstor.com>
This file is part of RockStor.
```
with:
```
Copyright (joint work) 2024 The Rockstor Project <https://rockstor.com>
```

---

Associated Copyright references/updates elsewhere in the project:
- Website: "Copyright © 2024 The Rockstor Project" via: https://github.com/rockstor/rockstor-website/pull/93
- Installer (EULA): "Copyright © 2012-2023 The Rockstor Project" see: https://rockstor.com/eula.html via: sed edits to our upstream performed in: https://github.com/rockstor/rockstor-installer/blob/master/config.sh

**Comment by FroggyFlox** [comment]
Thanks a lot for all of this, @phillxnet !

I agree with you and think we should go ahead and proceed with the proposal you detailed above.

**Comment by Hooverdan96** [comment]
I fully agree with these changes to the Copyright Notices. @phillxnet, thanks for digging into this, it's not the most exciting activity, but necessary.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2892
